lib/compress.js

```diff
@@ -261,6 +262,24 @@ merge(Compressor.prototype, {
                         d.fixed = false;
                     }
                 }
+                if (node instanceof AST_Defun) {
+                    var d = node.name.definition();
+                    if (!toplevel && d.global || is_safe(d)) {
+                        d.fixed = false;
+                    } else {
+                        // Make a shallow copy of definition body but as
+                        // expression instead - ensure to clone whatever
+                        // child nodes you extract from this
```

Comment:
I'd recommend a deep copy rather than a shallow one.

It's the last thing I'd expect in a tree optimization that it could potentially have cycles.